test/framework/workload.go

```diff
@@ -21,3 +32,60 @@ func (w *WorkloadCluster) DeleteCluster(opts ...CommandOpt) {
 	opts = append(opts, withKubeconfig(w.ManagementClusterKubeconfigFile()))
 	w.deleteCluster(opts...)
 }
+
+// ApplyClusterManifest uses client-side logic to create/update objects defined in a cluster yaml manifest.
+func (w *WorkloadCluster) ApplyClusterManifest() {
+	ctx := context.Background()
+	w.T.Logf("Applying workload cluster %s spec located at %s", w.ClusterName, w.ClusterConfigLocation)
+	w.applyClusterManifest(ctx)
+	w.waitForKubeconfig(ctx)
+	w.ValidateClusterState()
```

Comment:
I would prefer if we called this separately. IMO this method should just call kubectl apply. If that succeeds, return. And leave the wait and validations for the test to decide.

In general, the framework should offer exported methods for individual operations and leave the orchestration of those operations to the tests.